app/models/species.rb

```diff
@@ -13,4 +13,15 @@ class Specie < ApplicationRecord
 
   validates :wound_threshold, :strain_threshold, numericality: { only_integer: true, greater_than_or_equal_to: 0,
                                                                  less_than_or_equal_to: 99 }
-end
\ No newline at end of file
+
+  scope :with_name, ->(name) { where(name: name) }
+
+  def self.upload(json_file)
+    JsonSpecies.to_a(json_file).each do |json_specie|
+      json_specie_attrs = json_specie.attributes
+      Species.with_name(json_specie.name).first_or_create(json_specie_attrs).update(json_specie_attrs)
+    end
+  end
```

Comment:
If I understand this method correctly, you're passing it a `json_file`, iterating over each element and creating a `Specie` with json attributes, right?

This sounds like a good opportunity to create a service class for this

```ruby
# Under services folder 

class SpeciesUploder
  
   def initialize json_file
      @json_file = @json_file
    end

    def run
      # all your code in here
    end
end
```

If you think about it `upload` method should not be inside `Specie` model, since it does not add any business value, besides moving it into a service class will make our code more maintainable and easy to test